loadtest/store/memstore/store_test.go

```diff
@@ -71,6 +71,39 @@ func TestUser(t *testing.T) {
 		require.Equal(t, p, uu)
 	})
 
+	t.Run("SetPosts", func(t *testing.T) {
+		err := s.SetPosts(nil)
+		require.Error(t, err)
+		err = s.SetPosts([]*model.Post{})
+		require.Error(t, err)
+		p := []*model.Post{&model.Post{Id: model.NewId()}}
+		err = s.SetPosts(p)
+		require.NoError(t, err)
+		uu, err := s.Post(p[0].Id)
+		require.NoError(t, err)
+		require.Equal(t, p[0], uu)
+	})
+
+	t.Run("ChannelPosts", func(t *testing.T) {
+		channelId := model.NewId()
```

Comment:
Question: Is there a way to have a `AllTest` method or handler to create a new instance of the storage in every test `s := New()` instead at the beginning of the method?

Reply:
You mean before every subtests? If so I am not sure you can. 
We could probably leverage [this](https://golang.org/pkg/testing/#hdr-Main) to init the store just once before every test but since the cost of creating a new *in memory* store is not that great I don't believe the extra complexity is justified.

Reply:
I tried in a few days ago to have that running but I wasn't able. I'll try again.

The problem is not the performance cost but to have a clean state in every test. In some storage methods we're appending and not replacing the data and we could find that data which is not related with our current test in memory

Reply:
> I tried in a few days ago to have that running but I wasn't able. I'll try again.
> 
> The problem is not the performance cost but to have a clean state in every test. In some storage methods we're appending and not replacing the data and we could find that data which is not related with our current test in memory

Yes, that depends on a test to test basis but normally we want to have an empty state. Were you having issues creating a new store inside the subtest?

Reply:
I did it [here](https://github.com/mattermost/mattermost-load-test-ng/blob/master/loadtest/store/memstore/store_test.go#L186) and was the way to do it. Creating a new store in every test

Reply:
Looks good to me :+1: